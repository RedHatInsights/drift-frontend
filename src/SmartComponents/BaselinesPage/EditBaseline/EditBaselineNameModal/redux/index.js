export { default as editNameModalActions } from './actions';
