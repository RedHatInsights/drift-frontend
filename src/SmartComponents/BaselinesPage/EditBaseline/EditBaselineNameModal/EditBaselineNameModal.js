import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Button, Form, FormGroup, Modal, TextInput } from '@patternfly/react-core';

import { editBaselineActions } from '../redux';

export class EditBaselineNameModal extends Component {
    constructor(props) {
        super(props);

        /*eslint-disable camelcase*/
        this.state = {
            baselineName: this.props.baselineData.display_name
        };
        /*eslint-enable camelcase*/

        this.updateBaselineName = (value) => {
            this.setState({ baselineName: value });
        };

        this.confirmModal = this.confirmModal.bind(this);
    }

    async confirmModal() {
        const { baselineName } = this.state;
        const { baselineData, patchBaseline, toggleEditNameModal } = this.props;

        try {
            /*eslint-disable camelcase*/
            await patchBaseline(baselineData.id, { display_name: baselineName, facts_patch: []});
            /*eslint-enable camelcase*/

            toggleEditNameModal();
        } catch (e) {
            // do nothing and let redux handle
        }
    }

    cancelModal = () => {
        const { toggleEditNameModal, baselineData } = this.props;

        this.updateBaselineName(baselineData.display_name);
        toggleEditNameModal();
    }

    checkKeyPress = (event) => {
        if (event.key === 'Enter') {
            event.preventDefault();
            this.confirmModal();
        }
    }

    renderModalBody() {
        const { baselineName } = this.state;
        const { error } = this.props;

        return (<div className='fact-value'>
            <Form>
                <FormGroup
                    label='Baseline title'
                    isRequired
                    fieldId='baseline name'
                    helperTextInvalid={ error.hasOwnProperty('detail') ? error.detail : null }
                    isValid={ !error.hasOwnProperty('status') }
                    onKeyPress={ this.checkKeyPress }
                >
                    <TextInput
                        value={ baselineName }
                        type="text"
                        onChange={ this.updateBaselineName }
                        isValid={ !error.hasOwnProperty('status') }
                        aria-label="baseline name"
                    />
                </FormGroup>
            </Form>
        </div>);
    }

    render() {
        const { modalOpened } = this.props;

        return (
            <Modal
                className='small-modal-body'
                title="Edit name"
                isOpen={ modalOpened }
                onClose={ this.cancelModal }
                isFooterLeftAligned
                actions={ [
                    <Button
                        key="confirm"
                        variant="primary"
                        onClick={ this.confirmModal }>
                        Save
                    </Button>,
                    <Button
                        key="cancel"
                        variant="secondary"
                        onClick={ this.cancelModal }>
                        Cancel
                    </Button>
                ] }
            >
                { this.renderModalBody() }
            </Modal>
        );
    };
}

EditBaselineNameModal.propTypes = {
    baselineData: PropTypes.object,
    toggleEditNameModal: PropTypes.func,
    modalOpened: PropTypes.bool,
    patchBaseline: PropTypes.func,
    error: PropTypes.object
};

function mapDispatchToProps(dispatch) {
    return {
        patchBaseline: (baselineId, newBaselineBody) => dispatch(editBaselineActions.patchBaseline(baselineId, newBaselineBody))
    };
}

export default connect(null, mapDispatchToProps)(EditBaselineNameModal);
