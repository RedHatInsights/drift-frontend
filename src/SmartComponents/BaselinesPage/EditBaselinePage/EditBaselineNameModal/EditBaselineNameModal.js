import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Button, Form, FormGroup, Modal, ModalVariant, TextInput, ValidatedOptions, Alert } from '@patternfly/react-core';

import { editBaselineActions } from '../redux';

export class EditBaselineNameModal extends Component {
    constructor(props) {
        super(props);

        /*eslint-disable camelcase*/
        this.state = {
            baselineName: this.props.baselineData.display_name
        };
        /*eslint-enable camelcase*/

        this.updateBaselineName = (value) => {
            this.setState({ baselineName: value });
        };

        this.confirmModal = this.confirmModal.bind(this);
    }

    async confirmModal() {
        const { baselineName } = this.state;
        const { baselineData, patchBaseline, toggleEditNameModal } = this.props;

        try {
            /*eslint-disable camelcase*/
            await patchBaseline(baselineData.id, { display_name: baselineName, facts_patch: []});
            /*eslint-enable camelcase*/

            toggleEditNameModal();
        } catch (e) {
            // do nothing and let redux handle
        }
    }

    cancelModal = () => {
        const { toggleEditNameModal, baselineData } = this.props;

        this.updateBaselineName(baselineData.display_name);
        toggleEditNameModal();
    }

    checkKeyPress = (event) => {
        if (event.key === 'Enter') {
            event.preventDefault();
            this.confirmModal();
        }
    }

    renderModalBody() {
        const { baselineName } = this.state;
        const { error } = this.props;
        const hasError = Object.prototype.hasOwnProperty.call(error, 'status')  ? ValidatedOptions.error : null;

        return (<div className='fact-value'>
            <Form>
                <FormGroup
                    label='Baseline title'
                    isRequired
                    fieldId='baseline name'
                    helperTextInvalid={ Object.prototype.hasOwnProperty.call(error, 'detail') ? error.detail : null }
                    validated={ hasError }
                    onKeyPress={ this.checkKeyPress }
                >
                    <TextInput
                        value={ baselineName }
                        type="text"
                        onChange={ this.updateBaselineName }
                        validated={ hasError }
                        aria-label="baseline name"
                    />
                </FormGroup>
            </Form>
        </div>);
    }

    render() {
        const { modalOpened, error } = this.props;

        return (
            <Modal
                className="drift"
                variant={ ModalVariant.small }
                title="Edit name"
                isOpen={ modalOpened }
                onClose={ this.cancelModal }
                actions={ [
                    <Button
                        key="confirm"
                        variant="primary"
                        ouiaId="save"
                        onClick={ this.confirmModal }>
                        Save
                    </Button>,
                    <Button
                        key="cancel"
                        variant="link"
                        ouiaId="cancel"
                        onClick={ this.cancelModal }>
                        Cancel
                    </Button>
                ] }
            >
                { error.status && <Alert
                    variant='danger'
                    isInline
                    title={ 'Status: ' + error.status }
                >
                    <p>
                        { error.detail }
                    </p>
                </Alert> }
                { this.renderModalBody() }
            </Modal>
        );
    };
}

EditBaselineNameModal.propTypes = {
    baselineData: PropTypes.object,
    toggleEditNameModal: PropTypes.func,
    modalOpened: PropTypes.bool,
    patchBaseline: PropTypes.func,
    error: PropTypes.object
};

function mapDispatchToProps(dispatch) {
    return {
        patchBaseline: (baselineId, newBaselineBody) => dispatch(editBaselineActions.patchBaseline(baselineId, newBaselineBody))
    };
}

export default connect(null, mapDispatchToProps)(EditBaselineNameModal);
