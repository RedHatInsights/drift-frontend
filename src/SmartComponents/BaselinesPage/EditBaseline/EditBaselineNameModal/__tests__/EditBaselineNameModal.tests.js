import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { EditBaselineNameModal } from '../EditBaselineNameModal';
import editBaselineFixtures from '../../__tests__/helpers.fixtures';

describe('EditBaselineNameModal', () => {
    let props;

    beforeEach(() => {
        props = {
            baselineData: editBaselineFixtures.mockBaselineData1,
            modalOpened: true,
            error: {}
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <EditBaselineNameModal { ...props }/>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render mount correctly', () => {
        const wrapper = mount(
            <EditBaselineNameModal { ...props }/>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    describe('API', () => {
        it('should update baseline name', () => {
            const wrapper = shallow(
                <EditBaselineNameModal { ...props }/>
            );

            wrapper.find('ForwardRef').simulate('change', 'new-lotr-baseline');
            expect(wrapper.state('baselineName')).toEqual('new-lotr-baseline');
        });

        it('should confirm modal with patch', () => {
            const patchBaseline = jest.fn();
            const wrapper = mount(
                <EditBaselineNameModal
                    { ...props }
                    patchBaseline={ patchBaseline }
                />
            );

            wrapper.find('.pf-c-button').at(1).simulate('click');
            expect(patchBaseline).toHaveBeenCalled();
        });

        it('should confirm modal with error', () => {
            props.error = { detail: 'this is an error', status: 400 };
            const wrapper = mount(
                <EditBaselineNameModal
                    { ...props }
                />
            );

            wrapper.find('.pf-c-button').at(1).simulate('click');
            expect(wrapper.find('FormGroup').prop('helperTextInvalid')).toEqual('this is an error');
        });

        it('should cancel modal', () => {
            const toggleEditNameModal = jest.fn();
            const wrapper = mount(
                <EditBaselineNameModal
                    { ...props }
                    toggleEditNameModal={ toggleEditNameModal }
                />
            );

            wrapper.find('.pf-c-button').at(0).simulate('click');
            expect(toggleEditNameModal).toHaveBeenCalled();
        });

        it('should catch key press', () => {
            const patchBaseline = jest.fn();
            const wrapper = shallow(
                <EditBaselineNameModal
                    { ...props }
                    patchBaseline={ patchBaseline }
                />
            );

            wrapper.find('FormGroup').simulate('keypress', { key: 'Enter', preventDefault: jest.fn() });
            expect(patchBaseline).toHaveBeenCalled();
        });
    });
});
