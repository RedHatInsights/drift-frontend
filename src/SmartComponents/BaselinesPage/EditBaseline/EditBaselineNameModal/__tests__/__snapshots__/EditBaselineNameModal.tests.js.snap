// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditBaselineNameModal should render correctly 1`] = `
<Modal
  actions={
    Array [
      <Unknown
        onClick={[Function]}
        variant="primary"
      >
        Save
      </Unknown>,
      <Unknown
        onClick={[Function]}
        variant="link"
      >
        Cancel
      </Unknown>,
    ]
  }
  appendTo={<body />}
  ariaDescribedById=""
  className="small-modal-body"
  hideTitle={false}
  isFooterLeftAligned={true}
  isLarge={false}
  isOpen={true}
  isSmall={false}
  onClose={[Function]}
  showClose={true}
  title="Edit name"
>
  <div
    className="fact-value"
  >
    <Form>
      <FormGroup
        fieldId="baseline name"
        helperTextInvalid={null}
        isRequired={true}
        isValid={true}
        label="Baseline title"
        onKeyPress={[Function]}
      >
        <ForwardRef
          aria-label="baseline name"
          isValid={true}
          onChange={[Function]}
          type="text"
          value="lotr-baseline"
        />
      </FormGroup>
    </Form>
  </div>
</Modal>
`;

exports[`EditBaselineNameModal should render mount correctly 1`] = `
<EditBaselineNameModal
  baselineData={
    Object {
      "account": "12345678",
      "baseline_facts": Array [
        Object {
          "name": "Sauron",
          "value": "the Dark Lord",
        },
        Object {
          "name": "Galadriel",
          "value": "the Elven Queen",
        },
        Object {
          "name": "The Fellowship of the Ring",
          "values": Array [
            Object {
              "name": "Frodo",
              "value": "Baggins",
            },
            Object {
              "name": "Samwise",
              "value": "Gamgee",
            },
            Object {
              "name": "Gandalf",
              "value": "the Grey",
            },
            Object {
              "name": "Meriadoc",
              "value": "Brandybuck",
            },
            Object {
              "name": "Peregrin",
              "value": "Took",
            },
            Object {
              "name": "Gimli",
              "value": "son of Gloin",
            },
            Object {
              "name": "Legolas",
              "value": "Greenleaf",
            },
            Object {
              "name": "Boromir",
              "value": "son of Denethor",
            },
            Object {
              "name": "Aragorn",
              "value": "son of Arathorn",
            },
          ],
        },
      ],
      "created": "2019-10-17T16:20:06.050710Z",
      "display_name": "lotr-baseline",
      "fact_count": 3,
      "id": "jf8alskj-jf74-aje8-ke83-jf84ldjru439k",
      "updated": "2020-02-06T21:48:30.661510Z",
    }
  }
  error={Object {}}
  modalOpened={true}
>
  <Modal
    actions={
      Array [
        <Unknown
          onClick={[Function]}
          variant="primary"
        >
          Save
        </Unknown>,
        <Unknown
          onClick={[Function]}
          variant="link"
        >
          Cancel
        </Unknown>,
      ]
    }
    appendTo={
      <body
        class="pf-c-backdrop__open"
      >
        <div>
          <div
            class="pf-c-backdrop"
          >
            <div
              class="pf-l-bullseye"
            >
              <div
                aria-describedby="pf-modal-0"
                aria-label="Edit name"
                aria-modal="true"
                class="pf-c-modal-box small-modal-body"
                role="dialog"
              >
                <button
                  aria-label="Close"
                  class="pf-c-button pf-m-plain"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      transform=""
                    />
                  </svg>
                </button>
                <h1
                  class="pf-c-title pf-m-2xl"
                >
                   
                  Edit name
                   
                </h1>
                <div
                  class="pf-c-modal-box__body"
                  id="pf-modal-0"
                >
                  <div
                    class="fact-value"
                  >
                    <form
                      class="pf-c-form"
                      novalidate=""
                    >
                      <div
                        class="pf-c-form__group"
                      >
                        <label
                          class="pf-c-form__label"
                          for="baseline name"
                        >
                          <span
                            class="pf-c-form__label-text"
                          >
                            Baseline title
                          </span>
                          <span
                            aria-hidden="true"
                            class="pf-c-form__label-required"
                          >
                            *
                          </span>
                        </label>
                        <input
                          aria-invalid="false"
                          aria-label="baseline name"
                          class="pf-c-form-control"
                          type="text"
                          value="lotr-baseline"
                        />
                        
                      </div>
                    </form>
                  </div>
                </div>
                <div
                  class="pf-c-modal-box__footer pf-m-align-left"
                >
                  <button
                    class="pf-c-button pf-m-primary"
                    type="button"
                  >
                    Save
                  </button>
                  <button
                    class="pf-c-button pf-m-link"
                    type="button"
                  >
                    <span
                      class="pf-c-button__text"
                    >
                      Cancel
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </body>
    }
    ariaDescribedById=""
    className="small-modal-body"
    hideTitle={false}
    isFooterLeftAligned={true}
    isLarge={false}
    isOpen={true}
    isSmall={false}
    onClose={[Function]}
    showClose={true}
    title="Edit name"
  >
    <Portal
      containerInfo={
        <div>
          <div
            class="pf-c-backdrop"
          >
            <div
              class="pf-l-bullseye"
            >
              <div
                aria-describedby="pf-modal-0"
                aria-label="Edit name"
                aria-modal="true"
                class="pf-c-modal-box small-modal-body"
                role="dialog"
              >
                <button
                  aria-label="Close"
                  class="pf-c-button pf-m-plain"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      transform=""
                    />
                  </svg>
                </button>
                <h1
                  class="pf-c-title pf-m-2xl"
                >
                   
                  Edit name
                   
                </h1>
                <div
                  class="pf-c-modal-box__body"
                  id="pf-modal-0"
                >
                  <div
                    class="fact-value"
                  >
                    <form
                      class="pf-c-form"
                      novalidate=""
                    >
                      <div
                        class="pf-c-form__group"
                      >
                        <label
                          class="pf-c-form__label"
                          for="baseline name"
                        >
                          <span
                            class="pf-c-form__label-text"
                          >
                            Baseline title
                          </span>
                          <span
                            aria-hidden="true"
                            class="pf-c-form__label-required"
                          >
                            *
                          </span>
                        </label>
                        <input
                          aria-invalid="false"
                          aria-label="baseline name"
                          class="pf-c-form-control"
                          type="text"
                          value="lotr-baseline"
                        />
                        
                      </div>
                    </form>
                  </div>
                </div>
                <div
                  class="pf-c-modal-box__footer pf-m-align-left"
                >
                  <button
                    class="pf-c-button pf-m-primary"
                    type="button"
                  >
                    Save
                  </button>
                  <button
                    class="pf-c-button pf-m-link"
                    type="button"
                  >
                    <span
                      class="pf-c-button__text"
                    >
                      Cancel
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      }
    >
      <ModalContent
        actions={
          Array [
            <Unknown
              onClick={[Function]}
              variant="primary"
            >
              Save
            </Unknown>,
            <Unknown
              onClick={[Function]}
              variant="link"
            >
              Cancel
            </Unknown>,
          ]
        }
        ariaDescribedById=""
        className="small-modal-body"
        hideTitle={false}
        id="pf-modal-0"
        isFooterLeftAligned={true}
        isLarge={false}
        isOpen={true}
        isSmall={false}
        onClose={[Function]}
        showClose={true}
        title="Edit name"
      >
        <Backdrop>
          <div
            className="pf-c-backdrop"
          >
            <FocusTrap
              active={true}
              className="pf-l-bullseye"
              focusTrapOptions={
                Object {
                  "clickOutsideDeactivates": true,
                }
              }
              paused={false}
            >
              <div
                className="pf-l-bullseye"
              >
                <ModalBox
                  className="small-modal-body"
                  id="pf-modal-0"
                  isLarge={false}
                  isSmall={false}
                  style={Object {}}
                  title="Edit name"
                >
                  <div
                    aria-describedby="pf-modal-0"
                    aria-label="Edit name"
                    aria-modal="true"
                    className="pf-c-modal-box small-modal-body"
                    role="dialog"
                    style={Object {}}
                  >
                    <ModalBoxCloseButton
                      onClose={[Function]}
                    >
                      <Component
                        aria-label="Close"
                        className=""
                        onClick={[Function]}
                        variant="plain"
                      >
                        <ComponentWithOuia
                          component={[Function]}
                          componentProps={
                            Object {
                              "aria-label": "Close",
                              "children": <TimesIcon
                                color="currentColor"
                                noVerticalAlign={false}
                                size="sm"
                                title={null}
                              />,
                              "className": "",
                              "onClick": [Function],
                              "variant": "plain",
                            }
                          }
                          consumerContext={null}
                        >
                          <Button
                            aria-label="Close"
                            className=""
                            onClick={[Function]}
                            ouiaContext={
                              Object {
                                "isOuia": false,
                                "ouiaId": null,
                              }
                            }
                            variant="plain"
                          >
                            <button
                              aria-disabled={null}
                              aria-label="Close"
                              className="pf-c-button pf-m-plain"
                              disabled={false}
                              onClick={[Function]}
                              tabIndex={null}
                              type="button"
                            >
                              <TimesIcon
                                color="currentColor"
                                noVerticalAlign={false}
                                size="sm"
                                title={null}
                              >
                                <svg
                                  aria-hidden={true}
                                  aria-labelledby={null}
                                  fill="currentColor"
                                  height="1em"
                                  role="img"
                                  style={
                                    Object {
                                      "verticalAlign": "-0.125em",
                                    }
                                  }
                                  viewBox="0 0 352 512"
                                  width="1em"
                                >
                                  <path
                                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                                    transform=""
                                  />
                                </svg>
                              </TimesIcon>
                            </button>
                          </Button>
                        </ComponentWithOuia>
                      </Component>
                    </ModalBoxCloseButton>
                    <ModalBoxHeader
                      hideTitle={false}
                    >
                      <Title
                        className=""
                        headingLevel="h1"
                        size="2xl"
                      >
                        <h1
                          className="pf-c-title pf-m-2xl"
                        >
                           
                          Edit name
                           
                        </h1>
                      </Title>
                    </ModalBoxHeader>
                    <ModalBoxBody
                      id="pf-modal-0"
                    >
                      <div
                        className="pf-c-modal-box__body"
                        id="pf-modal-0"
                      >
                        <div
                          className="fact-value"
                        >
                          <Form>
                            <form
                              className="pf-c-form"
                              noValidate={true}
                            >
                              <FormGroup
                                fieldId="baseline name"
                                helperTextInvalid={null}
                                isRequired={true}
                                isValid={true}
                                label="Baseline title"
                                onKeyPress={[Function]}
                              >
                                <div
                                  className="pf-c-form__group"
                                  onKeyPress={[Function]}
                                >
                                  <label
                                    className="pf-c-form__label"
                                    htmlFor="baseline name"
                                  >
                                    <span
                                      className="pf-c-form__label-text"
                                    >
                                      Baseline title
                                    </span>
                                    <span
                                      aria-hidden="true"
                                      className="pf-c-form__label-required"
                                    >
                                      *
                                    </span>
                                  </label>
                                  <ForwardRef
                                    aria-label="baseline name"
                                    isValid={true}
                                    onChange={[Function]}
                                    type="text"
                                    value="lotr-baseline"
                                  >
                                    <TextInputBase
                                      aria-label="baseline name"
                                      className=""
                                      innerRef={null}
                                      isDisabled={false}
                                      isReadOnly={false}
                                      isRequired={false}
                                      isValid={true}
                                      onChange={[Function]}
                                      type="text"
                                      validated="default"
                                      value="lotr-baseline"
                                    >
                                      <input
                                        aria-invalid={false}
                                        aria-label="baseline name"
                                        className="pf-c-form-control"
                                        disabled={false}
                                        onChange={[Function]}
                                        readOnly={false}
                                        required={false}
                                        type="text"
                                        value="lotr-baseline"
                                      />
                                    </TextInputBase>
                                  </ForwardRef>
                                </div>
                              </FormGroup>
                            </form>
                          </Form>
                        </div>
                      </div>
                    </ModalBoxBody>
                    <ModalBoxFooter
                      isLeftAligned={true}
                    >
                      <div
                        className="pf-c-modal-box__footer pf-m-align-left"
                      >
                        <Component
                          key="confirm"
                          onClick={[Function]}
                          variant="primary"
                        >
                          <ComponentWithOuia
                            component={[Function]}
                            componentProps={
                              Object {
                                "children": "Save",
                                "onClick": [Function],
                                "variant": "primary",
                              }
                            }
                            consumerContext={null}
                          >
                            <Button
                              onClick={[Function]}
                              ouiaContext={
                                Object {
                                  "isOuia": false,
                                  "ouiaId": null,
                                }
                              }
                              variant="primary"
                            >
                              <button
                                aria-disabled={null}
                                aria-label={null}
                                className="pf-c-button pf-m-primary"
                                disabled={false}
                                onClick={[Function]}
                                tabIndex={null}
                                type="button"
                              >
                                Save
                              </button>
                            </Button>
                          </ComponentWithOuia>
                        </Component>
                        <Component
                          key="cancel"
                          onClick={[Function]}
                          variant="link"
                        >
                          <ComponentWithOuia
                            component={[Function]}
                            componentProps={
                              Object {
                                "children": "Cancel",
                                "onClick": [Function],
                                "variant": "link",
                              }
                            }
                            consumerContext={null}
                          >
                            <Button
                              onClick={[Function]}
                              ouiaContext={
                                Object {
                                  "isOuia": false,
                                  "ouiaId": null,
                                }
                              }
                              variant="link"
                            >
                              <button
                                aria-disabled={null}
                                aria-label={null}
                                className="pf-c-button pf-m-link"
                                disabled={false}
                                onClick={[Function]}
                                tabIndex={null}
                                type="button"
                              >
                                <span
                                  className="pf-c-button__text"
                                >
                                  Cancel
                                </span>
                              </button>
                            </Button>
                          </ComponentWithOuia>
                        </Component>
                      </div>
                    </ModalBoxFooter>
                  </div>
                </ModalBox>
              </div>
            </FocusTrap>
          </div>
        </Backdrop>
      </ModalContent>
    </Portal>
  </Modal>
</EditBaselineNameModal>
`;
